.github/workflows/patroni_backup.yaml

```diff
@@ -0,0 +1,182 @@
+name: Deploy Patroni(Postgres) backup containers
+
+on:
+  workflow_dispatch:
+
+jobs:
+  # build the backup patroni container
+  build:
+    name: 'Build Patroni Backup APP'
+    runs-on: ubuntu-latest
+    env:
+      BUILD_ID: ${{ github.event.number }}
+      BUILD_NAMESPACE: d89793-tools
+      NAME: patroni-backup
+      BUILD_TAG: latest
+    steps:
+      # Checkout the PR branch
+      - name: Print env
+        run: |
+          echo BUILD ID: $BUILD_ID
+          echo BUILD NAMESPACE: $BUILD_NAMESPACE
+      
+      - name: Checkout Target Branch
+        uses: actions/checkout@v1
+      
+      # Log in to OpenShift.
+      # Note: The secrets needed to log in are NOT available if the PR comes from a FORK.
+      # PR's must originate from a branch off the original repo or else all openshift `oc` commands will fail.
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
+
+      # Build backup container
+      - name: Build backup image
+        working-directory: './deployment/openshift/databases'
+        run: |
+          oc -n ${BUILD_NAMESPACE} process -f prebuilt-db-backup-build.yaml \
+          -p NAME=${NAME} \
+          -p TAG=${BUILD_TAG} | oc -n ${BUILD_NAMESPACE} apply -f - --wait
+
+  # deploy the backup image in Dev
+  deployDev:
```

Comment:
I really don't like this one-job-per-env pattern. Not very DRY. Also you are missing the `if: ${{!github.event.inputs.environment || github.event.inputs.environment == 'dev'}}` guards that the deploy.yml workflow has.

Could you see about merging these into a single `deploy` job and just template in the `github.event.inputs.environment` variable where required to get env-specific behaviour?

Reply:
Missed this comment before requesting the review again. Kindly hold off until this is resolved.
----

ahh, sure! we can add environment checks - in fact we should 💯 

Also, agree regarding the single`deploy` job, but I was just following the suit there. Will give it a try and get back.

Reply:
@will-freshworks - updated PR to make deploy step generic; 

The core of the issue got solved by just naming input to "prod" instead of "production" 🤷‍♂️ ✅